library/src/test/java/com/datatorrent/lib/projection/ProjectionTest.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.projection;
+
+import java.lang.reflect.Field;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests for ProjectionOperator
+ */
+
+public class ProjectionTest
+{
+  private static final Logger logger = LoggerFactory.getLogger(ProjectionTest.class);
+
+  public static class DummyPOJO
+  {
+    private long projected;
+    private long remainder;
+
+    public long getProjected()
+    {
+      return projected;
+    }
+
+    public void setProjected(long _projected)
+    {
+      projected = _projected;
```

Comment:
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html
The convention, is to always begin your variable names with a letter, not "$" or "_".

this.projected = projected is more popular way of doing it.